src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1047,6 +1047,66 @@ internal void OnAdditionalDocumentTextUpdatedOnDisk(DocumentId documentId)
             return this.ServiceProvider.GetService(typeof(TService)) as TInterface;
         }
 
+        internal override bool CanAddProjectReference(ProjectId referencingProject, ProjectId referencedProject)
+        {
+            _foregroundObject.AssertIsForeground();
+
+            IVsHierarchy referencingHierarchy;
+            IVsHierarchy referencedHierarchy;
+            if (!TryGetHierarchy(referencingProject, out referencingHierarchy) ||
+                !TryGetHierarchy(referencedProject, out referencedHierarchy))
+            {
+                return false;
+            }
+
+            var referencingProjectFlavor3 = referencingHierarchy as IVsProjectFlavorReferences3;
+            var referencedProjectFlavor3 = referencedHierarchy as IVsProjectFlavorReferences3;
+
+            if (referencingProjectFlavor3 != null && referencedProjectFlavor3 != null)
```

Comment:
Ok.  I can def do that.  Note though that we should never get C++ projects.  These are always Roslyn projects that we're dealing with.  But even so, i'll rework the code to properly handle this issue.